packages/protocol/contracts/tokenvault/erc20/adapters/BaseAdapter.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: MIT
+//  _____     _ _         _         _
+// |_   _|_ _(_) |_____  | |   __ _| |__ ___
+//   | |/ _` | | / / _ \ | |__/ _` | '_ (_-<
+//   |_|\__,_|_|_\_\___/ |____\__,_|_.__/__/
+
+pragma solidity ^0.8.20;
+
+import "../../../common/EssentialContract.sol";
+
+/// @notice We need to implement the 'relayer functions' - which will forward the calls the
+/// ERC20Vault calls towards the L2 predeployed native contracts.
+interface IERC20TokenVaultRelayer {
+    /// @notice Mints `amount` tokens and assigns them to the `account` address.
+    /// @param token The erc20 token contract.
+    /// @param account The account to receive the minted tokens.
+    /// @param amount The amount of tokens to mint.
+    function mint(address token, address account, uint256 amount) external;
+
+    /// @notice Burns `amount` tokens from the `from` address.
+    /// @param token The erc20 token contract.
+    /// @param from The account from which the tokens will be burned.
+    /// @param amount The amount of tokens to burn.
+    function burn(address token, address from, uint256 amount) external;
+}
+
+/// @title BaseAdapter
+/// @notice It serves as a wrapper between the deployed USDC and the ERC20Vault - an extra layer for
+/// flexibility. It is not an ERC20 contract, but we need to implement the interfaces the ERC20Vault
+/// calls and relay over to other (native) contracts. We do not even need this contract to be
+/// initialized, because we need the implementation of it, the EssentialContract is only because of
+/// storage compatibility with ERC20Vault.
+abstract contract BaseAdapter is EssentialContract, IERC20TokenVaultRelayer {
+    uint256[50] private __gap;
```

Comment:
Does it need to be a `EssentialContract`? Seems safer to be not to have `uint256[50] private __gap;` in here of do any storage writes.